app/services/target_service.rb

```diff
@@ -8,6 +8,10 @@ def create!
     targets.create!(@params)
   end
 
+  def destroy!
+    targets.find(@params).destroy!
+  end
```

Comment:
I don't like the idea of having the same variable `@params` to be used for different cases, and different formats (when destroying, its just an id, and when creating its a full list of attrributes).
Feels difficult to debug and understand (you may know what is the method called at that moment to infer the format/value that `@params` will have)
I remember I suggested you with this approach or the alternative one with a service for each method. wdyt about moving to different services?
another alternative can be removing params from the initializer, but sending the required params on each method itself, like:
```suggestion
  def destroy!(id)
    targets.find(id).destroy!
  end
```
(and same change for the create: `def create!(create_params)`)
wdyt?

Reply:
I understand your point, but I don´t know what is the best implementation. wdyt? Because I like the idea of send the params in the methods, I think I can reuse some code of the service for each method, but I don´t know if eventualy the service can become too complicated to understand.

Reply:
I had the answer for the beginning and you will probably hate me for that 😅

I think the right approach here is to not have a service at all, because using a service pattern here you are just calling the ActiveRecord methods that already do that! Create and destroy (and in future update maybe). I liked your way of thinking about using services, but in this particular case where you are just calling the existing active record methods without any extra logic, its adding unecessary complexity.
If the case is that you are transforming data, or creating extra records or whathever, then it makes sense to have a service to encapsulate that logic and not having too much code in the controller. But here you can just call destroy and create from the controller and everything will be fine and the code will be clean 😉 (if on next tickets you are adding extra logic to some of the controller actions, then yes, a service will be better and you can refactor it, but not at this stage)

Reply:
Ok! I understand you, I thought that a service was be a good approach because it facilitates the expansion, but I will heed the voice of experience and I will change the implementation.